components/movementsensor/imuwit/imu.go

```diff
@@ -184,8 +168,21 @@ func NewWit(
 	}
 
 	options.PortName = conf.Port
-	if conf.BaudRate > 0 {
+
+	// Validating baud rate
+	isValid := false
+	for _, val := range baudRateList {
+		if val == conf.BaudRate {
+			isValid = true
+		}
+	}
+
+	if conf.BaudRate > 0 && isValid {
```

Comment:
[q] how do we feel about moving the validation logic here? This makes the field truly optional in configs.

Reply:
I think this is ok, baudRate should be optional anyway. Only thing I can consider is wether 9600 is always a valid default?

Reply:
it's not
<img width="476" alt="Screenshot 2023-02-17 at 09 41 41" src="https://user-images.githubusercontent.com/35934754/219685529-e661ae04-c8c1-4770-ad07-68202badece8.png">

 for the BWT61CL and the BWT90, they only allow a baudrate of 115200. This is true for wit's Bluetooth sensor series.
 for  the HWT901B TTL and what seems like other TTL/wire series, baud rate selection is allowed between 115200 and 9600.
 
 I'll add these manuals to the top of the file, and change the default baudrate to 115200

Reply:
just a thought here, we might want to add logic to some drivers to try out different baudrates when none are set before giving up

Reply:
that would be a cool improvement, made a ticket to add it as a utility and decide if we do this for all sensors.
 https://viam.atlassian.net/browse/RSDK-1991?atlOrigin=eyJpIjoiNGI1NzQ2NTMyMTQ3NDkzOWI2MGRhMmQ3ZDk4YWEwNDAiLCJwIjoiaiJ9

Reply:
@penguinland your proposed change will produce the exact same behavior as the state of this code right now. So I'd prefer not to change it back. But if you feel that strongly about it, I will.

Reply:
That field is omitted empty anyway. So not checking in in the Validate function and setting a default is actually more expected.

Reply:
Changing back to the old Vlaidation logic